com.unity.ml-agents/Runtime/Academy.cs

```diff
@@ -125,7 +125,7 @@ public bool IsCommunicatorOn
         List<ModelRunner> m_ModelRunners = new List<ModelRunner>();
 
         // Flag used to keep track of the first time the Academy is reset.
-        bool m_FirstAcademyReset;
+        internal bool FirstAcademyReset;
```

Comment:
Can we call this variable `FirstAcademyResetHappened` or `HadFirstReset`? The logic is confusing since it's `false` until set `true` but when it's `true` it's NOT the first reset.

Reply:
Maybe just check `Academt.TotalStepCount == 0`? they should be equivalent, right?

Reply:
Done!